pandas/_libs/lib.pyx

```diff
@@ -2550,6 +2576,26 @@ def maybe_convert_objects(ndarray[object] objects,
     return objects
 
 
+cdef _infer_all_nats(dtype, ndarray datetimes, ndarray timedeltas):
+    """
```

Comment:
can you type dtype & narrow the types on datetimes/timedeltas if possible.

also a return type ?

Reply:
not really, as there is no good way to declare `ndarray[dt64]`, and both the dtype arg and return type aren't representable here